x-pack/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -0,0 +1,365 @@
+    index: Object.values(indexPatterns),
+    size: 0,
+    ignoreUnavailable: true,
+    filterPath: [
+      'aggregations.indices.buckets'
+    ],
+    body: {
+      query: {
+        bool: {
+          filter: filters,
+        }
+      },
+      aggs: {
+        indices: {
+          terms: {
+            field: '_index',
+            size: 50,
+          },
+          aggs: {
+            es_uuids: {
+              terms: {
+                field: 'node_stats.node_id'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            kibana_uuids: {
+              terms: {
+                field: 'kibana_stats.kibana.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            },
+            beats_uuids: {
+              terms: {
+                field: 'beats_stats.beat.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                },
+                beat_type: {
+                  terms: {
+                    field: 'beats_stats.beat.type'
+                  }
+                }
+              }
+            },
+            logstash_uuids: {
+              terms: {
+                field: 'logstash_stats.logstash.uuid'
+              },
+              aggs: {
+                by_timestamp: {
+                  max: {
+                    field: 'timestamp'
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  };
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+  return await callWithRequest(req, 'search', params);
+};
+
+async function detectProducts(req) {
+  const result = {
+    [KIBANA_SYSTEM_ID]: {
+      doesExist: true,
+    },
+    'elasticsearch': {
+      doesExist: true,
+    },
+    [BEATS_SYSTEM_ID]: {
+      mightExist: false,
+    },
+    [APM_CUSTOM_ID]: {
+      mightExist: false,
+    },
+    [LOGSTASH_SYSTEM_ID]: {
+      mightExist: false,
+    }
+  };
+
+  const msearch = [
+    { index: '*beat-*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: '.management-beats*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: 'logstash-*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: '.logstash*' },
+    { size: 0, terminate_after: 0 },
+
+    { index: 'apm*' },
+    { size: 0, terminate_after: 0 },
+  ];
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+  const {
+    responses: [
+      beatsDetectionResponse1,
+      beatsDetectionResponse2,
+      logstashDetectionResponse1,
+      logstashDetectionResponse2,
```

Comment:
Naming: for these variables that end in 1 and 2, it might be clearer to name them based on what they actually represent, e.g. `beatsDataDetectionResponse`, `beatsManagementDetectionResponse`, etc.